ui/src/views/mint/github-step/github-repo-configuration.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  Button,
+  Card,
+  Dropdown,
+  DropdownItem,
+  Form,
+  Grid,
+  Icon,
+  IconButton,
+  Stepper,
+} from '@/components';
+import React, { useState } from 'react';
+import { Mint } from '../mint.context';
+import { RepoRow } from './github-repository-selection';
+
+//TODO remove once it's integrated with GH login
+const branches: DropdownItem[] = [
+  {
+    label: 'master',
+    value: 'master',
+  },
+  {
+    label: 'develop',
+    value: 'develop',
+  },
+  {
+    label: 'feature/branch',
+    value: 'feature/branch',
+  },
+];
+
+export const GithubRepoConfiguration: React.FC = () => {
+  const {
+    repositoryName,
+    branchName,
+    commitHash,
+    setGithubStep,
+    setRepositoryConfig,
+  } = Mint.useContext();
+  const { nextStep } = Stepper.useContext();
+  const [branchSelected, setBranchSelected] = useState(branchName);
+  const [commitHashSelected, setCommitHashSelected] = useState(commitHash);
+
+  const handlePrevStepClick = () => {
+    setGithubStep(2);
+    setRepositoryConfig('', '');
+  };
+
+  const handleBranchChange = (dorpdownOption: DropdownItem) => {
+    //TODO we'll have to check the data that GH API returns
+    setBranchSelected(dorpdownOption.value);
+  };
+
+  const handleCommitHashChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setCommitHashSelected(e.target.value);
+  };
+
+  const handleContinueClick = () => {
+    setRepositoryConfig(branchSelected, commitHashSelected);
+    nextStep();
+  };
+
+  return (
+    <Card.Container css={{ width: '$107h' }}>
+      <Card.Heading

```

Comment:
This is not mandatory right now, but seems like this component is going to grow when adding the remote fetching part. Probably when we got there is a good idea to split it in two components here `Header` and `Body` or something like that, so we can scope out which handlers and functions are related to each part and this would be not too much complicated since you are using context you don't need to pass props.

Reply:
Ahh yeap, good catch! I was thinking to create a general card for the mint view cause I realize that we have the same style in all the card for the mint flow at least. But yeah, for now maybe I can split just this part